src/components/Input/DateInput.js

```diff
@@ -197,12 +198,15 @@ export default class DateInput extends React.Component {
     return this.state.value;
   };
 
-  getCurrentDate = () =>
-    parseValue(
-      this.props.value !== undefined ? this.props.value : this.state.value,
-      this.props.dateFormat,
-      this.props.parse
-    );
+  getCurrentDate = () => {
+    // If value prop is provided and not empty, use it
+    // Otherwise, use state.value if available, or fall back to initialCalendarDate
+    const valueToUse =
+      this.props.value !== undefined && this.props.value !== ''
+        ? this.props.value
+        : this.state.value || this.props.initialCalendarDate;
```

Comment:
I just realized `valueToUse` can be `undefined` whereas before it would always at least be an empty string.
```suggestion
        : this.state.value || this.props.initialCalendarDate || '';
```